src/objects.rs

```diff
@@ -333,6 +381,95 @@ impl ObjectData {
         Ok(ObjectShape::Polygon { points })
     }
 
+    fn new_text(attrs: Vec<OwnedAttribute>) -> Result<ObjectShape> {
+        let (
+            font_family,
+            pixel_size,
+            wrap,
+            color,
+            bold,
+            italic,
+            underline,
+            strikeout,
+            kerning,
+            halign,
+            valign,
+        ) = get_attrs!(
+            for v in attrs {
+                Some("fontfamily") => font_family = v,
+                Some("pixelsize") => pixel_size ?= v.parse(),
+                Some("wrap") => wrap ?= v.parse(),
+                Some("color") => color ?= v.parse(),
+                Some("bold") => bold ?= v.parse(),
+                Some("italic") => italic ?= v.parse(),
+                Some("underline") => underline ?= v.parse(),
+                Some("strikeout") => strikeout ?= v.parse(),
+                Some("kerning") => kerning ?= v.parse(),
+                Some("halign") => halign = match v.as_str() {
+                    "left" => HorizontalAlignment::Left,
+                    "center" => HorizontalAlignment::Center,
+                    "right" => HorizontalAlignment::Right,
+                    "justify" => HorizontalAlignment::Justify,
+                    _ => return Err(Error::MalformedAttributes("`halign` property did not contain a valid value of 'left', 'center', 'right' or 'justify'".to_string()))
+                },
+                Some("valign") => valign = match v.as_str() {
+                    "top" => VerticalAlignment::Top,
+                    "center" => VerticalAlignment::Center,
+                    "bottom" => VerticalAlignment::Bottom,
+                    _ => return Err(Error::MalformedAttributes(
+                        "`halign` property did not contain a valid value of 'top', 'center' or 'bottom'"
+                            .to_string(),
+                    )),
+                },
+            }
+            (
+
+                font_family,
+                pixel_size,
+                wrap,
+                color,
+                bold,
+                italic,
+                underline,
+                strikeout,
+                kerning,
+                halign,
+                valign,
+
+        )
+        );
+        let font_family = font_family.unwrap_or_else(|| "sans-serif".to_string());
+        let pixel_size = pixel_size.unwrap_or(16);
+        let wrap = if wrap == Some(1) { true } else { false };
+        let color = color.unwrap_or(Color {
+            red: 0,
+            green: 0,
+            blue: 0,
+            alpha: 255,
+        });
+        let bold = if bold == Some(1) { true } else { false };
```

Comment:
Why do we need `let bold = if bold == Some(1) { true } else { false }`? Could just `let bold = bold == Some(1)` suffice? Or `bold.contains(&1)`, heh.

Reply:
Oh lol yeah